src/benchmarks/micro/libraries/System.Memory/ReadOnlySpan.cs

```diff
@@ -13,25 +13,33 @@ namespace System.Memory
     public class ReadOnlySpan
     {
         private readonly string _sampeString = "this is a very nice sample string";
-        private readonly Consumer _consumer = new Consumer();
 
         [Benchmark]
         public ReadOnlySpan<char> StringAsSpan() => _sampeString.AsSpan();
         
         [Benchmark(OperationsPerInvoke = 16)]
-        public void GetPinnableReference()
+        public char GetPinnableReference()
         {
             ReadOnlySpan<char> span = _sampeString.AsSpan();
-            var consumer = _consumer;
-            
-            consumer.Consume(span.GetPinnableReference()); consumer.Consume(span.GetPinnableReference());
-            consumer.Consume(span.GetPinnableReference()); consumer.Consume(span.GetPinnableReference());
-            consumer.Consume(span.GetPinnableReference()); consumer.Consume(span.GetPinnableReference());
-            consumer.Consume(span.GetPinnableReference()); consumer.Consume(span.GetPinnableReference());
-            consumer.Consume(span.GetPinnableReference()); consumer.Consume(span.GetPinnableReference());
-            consumer.Consume(span.GetPinnableReference()); consumer.Consume(span.GetPinnableReference());
-            consumer.Consume(span.GetPinnableReference()); consumer.Consume(span.GetPinnableReference());
-            consumer.Consume(span.GetPinnableReference()); consumer.Consume(span.GetPinnableReference());
+            char c;
+
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
+            c = span.GetPinnableReference(); c = span.GetPinnableReference();
```

Comment:
is there a possibility that in the future JIT is going to replace all 32 calls with a single one?

@EgorBo or perhaps it can happen as of today with Mono using LLVM?

If so, we should most probably xor the results

```suggestion
            c = span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
            c ^= span.GetPinnableReference(); c ^= span.GetPinnableReference();
```

Reply:
@adamsitnik I've just checked - Mono-LLVM optimizes out all the dead stores (basicaly, does `span.GetPinnableReference()` only once). The "^=" trick doesn't help 🙂 (with the proposed change it optimized the whole function to `return 0`

Reply:
Regarding Ryujit I guess it's: https://github.com/dotnet/runtime/issues/13727

Reply:
@EgorBo thank you! Do you know how can we trick the Mono-LLVM ? Mix `+` with `^` ?

Reply:
No idea, `GetPinnableReference` is just too simple and returns a pointer. I guess only if you hide `span` to a non-inlineable method e.g. `GetSpan().GetPinnableReference() ^ ..` but I guess the overhead of that method will be bigger than the actual `GetPinnableReference`

Reply:
Weird. Surprisingly RYUJit doesn't eliminate dead stores (at least for this example). The code that I pasted above is precisely that gets generated today. I will investigate and possibly open an issue to track this.

Reply:
Perhaps we can do something like this? https://github.com/dotnet/performance/blob/8aed638c9ee65c034fe0cca4ea2bdc3a68d2a6b5/src/benchmarks/micro/libraries/System.Memory/Slice.cs#L293 
But yeah, that would have method call overhead too.

Reply:
In such case, I think that I am going to merge it as it is.

If it becomes a problem in the future, we might consider removing this benchmark because it looks like it might be impossible to properly benchmark this method (it's too simple).

/cc @billwert @DrewScoggins